src/snowflake/snowpark/session.py

```diff
@@ -831,19 +858,25 @@ def add_requirements(self, file_path: str) -> None:
             and the Snowflake server.
         """
         packages = []
-        with open(file_path) as f:
-            for line in f:
-                package = line.rstrip()
-                if package:
-                    packages.append(package)
-        self.add_packages(packages)
+        if file_path.endswith(".txt"):
+            with open(file_path) as f:
+                for line in f:
```

Comment:
it's still possible to have a text file without a `.txt` extension

Reply:
Updated